### PR TITLE
🚀♻️ Refactor Srcset

### PR DESCRIPTION
**Pull request by jridgewell**
Shaves a half kb off compiled size, and makes simplifies some of the algorithms.

### CODE REVIEW

File: src/srcset.js [hunk 1]
```diff
@@ -180,119 +165,110 @@ export class Srcset {
...
+        src += ` ${source.dpr}x`;
       }
+      res.push(src);
     }
     return res.join(', ');
   }
 }
 
 function sortByWidth(s1, s2) {
   user().assert(s1.width != s2.width, 'Duplicate width: %s', s1.width);
```
Comment (cathyxz) [inline src/srcset.js]: Is this sorting change intentional? 0.0

Comment (jridgewell) [reply src/srcset.js]: Yes, so we can stop backwards iterating.

File: src/srcset.js [hunk 2]
```diff
@@ -180,119 +165,110 @@ export class Srcset {
...
     return minIndex;
   }
 
   /**
-   * Returns the last source in the srcset, which is the default source.
-   * @return {!SrcsetSourceDef}
+   * Returns all URLs in the srcset.
+   * @return {!Array<string>}
    */
-  getLast() {
```
Comment (cathyxz) [inline src/srcset.js]: Welp, there's a test in `test-amp-image-lightbox.js` that uses this and needs to be updated too.

Comment (jridgewell) [reply src/srcset.js]: Done.

**Comment by jridgewell** [comment]
> Out of curiosity, what sparked this change?

Binary size reduction, and the algorithm always bothered me anyways. 😃

**Pull request merged**